### PR TITLE
[API-1790] Various Scene Tree bug fixes

### PR DESCRIPTION
**Pull request by danschultz**
## What

Fixes some timing issues during initialization that was causing the scene tree to not load. Should address issues found in the business intelligence and scene studio apps.

Also fixes an issue with getting the correct row from an event.

## Ticket

https://vertexvis.atlassian.net/browse/API-1790

### CODE REVIEW

File: packages/viewer/src/components/scene-tree/scene-tree.tsx [hunk 1]
```diff
@@ -480,11 +476,10 @@ export class SceneTree {
   @Method()
   public async getRowForEvent(event: MouseEvent | PointerEvent): Promise<Row> {
     const { clientY, currentTarget } = event;
-    const rowsEl = this.el.shadowRoot?.querySelector('.rows');
     if (
       currentTarget != null &&
-      rowsEl != null &&
-      getSceneTreeContainsElement(rowsEl, currentTarget as HTMLElement)
```
Comment (danschultz) [inline packages/viewer/src/components/scene-tree/scene-tree.tsx:487]: This was evaluating the incorrect element. When adding an event listener to the scene tree, `currentTarget` will be the scene tree and not the child that was clicked.

File: packages/viewer/src/components/scene-tree/scene-tree.tsx [hunk 2]
```diff
@@ -500,22 +495,43 @@ export class SceneTree {
...
+  protected disconnectedCallback(): void {
+    console.debug('Scene tree removed from DOM');
+
+    if (this.viewer != null) {
+      this.disconnectViewer(this.viewer);
+    }
+
+    if (this.controller != null) {
+      this.controller = undefined;
+    }
```
Comment (danschultz) [inline packages/viewer/src/components/scene-tree/scene-tree.tsx:519]: This removes any event listeners that were added.

File: packages/viewer/src/components/scene-tree/scene-tree.tsx [hunk 3]
```diff
@@ -500,22 +495,43 @@ export class SceneTree {
...
+  }
+
   protected componentWillLoad(): void {
+    const { sceneTreeHost } = this.getConfig().network;
+    this.client = new SceneTreeAPIClient(sceneTreeHost);
```
Comment (danschultz) [inline packages/viewer/src/components/scene-tree/scene-tree.tsx:524]: This needs to be set before `connectViewer` is called.

File: packages/viewer/src/components/scene-tree/scene-tree.tsx [hunk 4]
```diff
@@ -665,35 +683,21 @@ export class SceneTree {
     newViewer: HTMLVertexViewerElement | undefined,
     oldViewer: HTMLVertexViewerElement | undefined
   ): Promise<void> {
+    // StencilJS will invoke this callback even before the component has been
+    // loaded. According to their docs, this shouldn't happen. Return if the
+    // component hasn't been loaded.
+    // See https://stenciljs.com/docs/reactive-data#watch-decorator
+    if (!this.stateMap.componentLoaded) {
+      return;
+    }
```
Comment (danschultz) [inline packages/viewer/src/components/scene-tree/scene-tree.tsx:692]: This fixes an issue when a second scene tree gets initialized.

**Review by MadisonEhlers-Vertex — APPROVED** [review]
lgtm. Tested locally as well.

**Pull request merged**